### PR TITLE
Update API: JumpThreshold -> CartesianPrecision

### PR DESCRIPTION
**Pull request by rhaschke**
This should be merged only after the next MoveIt release.

### CODE REVIEW

File: core/src/solvers/cartesian_path.cpp [hunk 1]
```diff
@@ -54,7 +54,7 @@ CartesianPath::CartesianPath() {
 	auto& p = properties();
 	p.declare<geometry_msgs::PoseStamped>("ik_frame", "frame to move linearly (use for joint-space target)");
 	p.declare<double>("step_size", 0.01, "step size between consecutive waypoints");
-	p.declare<double>("jump_threshold", 1.5, "acceptable fraction of mean joint motion per step");
+	p.declare<moveit::core::CartesianPrecision>("precision", { 1e-3, 1e-2, 1e-4 }, "precision of linear path");
```
Comment (v4hn) [inline core/src/solvers/cartesian_path.cpp:57]: Why explicitly provide values here and not rely [on the defaults](https://github.com/moveit/moveit/blob/026c512b12e1723630deef9bf810066d7e95641c/moveit_core/robot_state/include/moveit/robot_state/cartesian_interpolator.h#L50-L55)? And why is the third value different from MoveIt's default of `1e-5`?

File: core/python/bindings/src/solvers.cpp [hunk 2]
```diff
@@ -99,6 +100,19 @@ void export_solvers(py::module& m) {
 	    .property<double>("max_step", "float: Limit any (single) joint change between two waypoints to this amount")
 	    .def(py::init<>());
 
+	py::class_<moveit::core::CartesianPrecision>(m, "CartesianPrecision", "precision for Cartesian interpolation")
+	    .def(py::init([](double translational, double rotational, double max_resolution) {
+		         return new moveit::core::CartesianPrecision{ translational, rotational, max_resolution };
+	         }),
+	         py::arg("translational") = 0.001, py::arg("rotational") = 0.01, py::arg("max_resolution") = 1e-5)
```
Comment (v4hn) [inline core/python/bindings/src/solvers.cpp:107]: I guess you could instantiate a default `CartesianPrecision` here and use the values from there?

File: core/include/moveit/task_constructor/solvers/cartesian_path.h [hunk 3]
```diff
@@ -57,7 +58,9 @@ class CartesianPath : public PlannerInterface
 	void setIKFrame(const std::string& link) { setIKFrame(Eigen::Isometry3d::Identity(), link); }
 
 	void setStepSize(double step_size) { setProperty("step_size", step_size); }
-	void setJumpThreshold(double jump_threshold) { setProperty("jump_threshold", jump_threshold); }
+	void setPrecision(const moveit::core::CartesianPrecision& precision) { setProperty("precision", precision); }
+	[[deprecated("Replace with setPrecision")]]  // clang-format off
```
Comment (v4hn) [inline core/include/moveit/task_constructor/solvers/cartesian_path.h:62]: Let's not provide a no-op function for something that was rather essential before. :neutral_face: 
People tend to ignore deprecations and still assume things will work.
How about a [`static_assert` on use](https://stackoverflow.com/a/18814056/21260084)?

**Review by v4hn — APPROVED** [review]
Looks good except for my few comments.

Do you really want to keep this lying here until a new moveit release?
I know we went for supporting released and master branch of moveit in the past, but I'm not sure I see the need for it.
Of course we should hold a new *release* tag until moveit is released as well.

File: core/include/moveit/task_constructor/solvers/cartesian_path.h (hunk 3 above)
Comment (rhaschke) [reply core/include/moveit/task_constructor/solvers/cartesian_path.h:62]: Somewhat ugly, but ok to enforce deprecation.

**Comment by v4hn** [comment]
Thanks for addressing the feedback. :+1: 

> Do you really want to keep this lying here until a new moveit release? I know we went for supporting released and master branch of moveit in the past, but I'm not sure I see the need for it. Of course we should hold a new release tag until moveit is released as well.

What's your plan for this? I would merge here if not for your statement above.

**Comment by rhaschke** [comment]
> What's your plan for this? I would merge here if not for your statement above.

I prefer to have a MoveIt release. Otherwise we enforce people to build from source.
I am happy to cut a new MoveIt release the next days, if there are no objections.

**Pull request merged**